internal/provider/resource_mackerel_channel.go

```diff
@@ -0,0 +1,255 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mackerelio/mackerel-client-go"
+)
+
+func resourceMackerelChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMackerelChannelCreate,
+		Read:   resourceMackerelChannelRead,
+		Update: resourceMackerelChannelUpdate,
+		Delete: resourceMackerelChannelDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Input value is limited, should have validate function.
https://github.com/hashicorp/terraform/blob/v0.13.5/helper/validation/validation.go#L130

```suggestion
			"type": {
				Type:     schema.TypeString,
				Required: true,
				ValidateFunc: validation.StringInSlice([]string{
					"email",
					"slack",
					"webhook",
				}, false),
			},
```

Reply:
[FYI]
Should be updated, when [mackerel api](https://mackerel.io/ja/api-docs/entry/channels#get) is updated to create other types such as `pagerduty`, etc.